CONTRIBUTING.rst

```diff
@@ -0,0 +1,480 @@
+
+    (c) The contribution was provided directly to me by some other
+        person who certified (a), (b) or (c) and I have not modified
+        it.
+
+    (d) I understand and agree that this project and the contribution
+        are public and that a record of the contribution (including
+        all personal information I submit with it, including my
+        sign-off) is maintained indefinitely and may be redistributed
+        consistent with this project or the open source license(s)
+        involved.
+
+DCO Sign-Off Methods
+====================
+
+The DCO requires a sign-off message in the following format appear on each
+commit in the pull request::
+
+   Signed-off-by: Zephyrus Zephyr <zephyrus@zephyrproject.org>
+
+The DCO text can either be manually added to your commit body, or you can add
+either ``-s`` or ``--signoff`` to your usual Git commit commands. If you forget
+to add the sign-off you can also amend a previous commit with the sign-off by
+running ``git commit --amend -s``. If you've pushed your changes to GitHub
+already you'll need to force push your branch after this with ``git push -f``.
+
+
+Prerequisites
+*************
+
+.. _Zephyr Project website: https://zephyrproject.org
+
+As a contributor, you'll want to be familiar with the Zephyr project, how to
+configure, install, and use it as explained in the `Zephyr Project website`_
+and how to set up your development environment as introduced in the Zephyr
+`Getting Started Guide`_.
+
+.. _Getting Started Guide:
+   https://www.zephyrproject.org/doc/getting_started/getting_started.html
+
+Linux is the preferred host environment for Zephyr development. As a Linux
+developer, you should be familiar with common tools such as Git and Make, and
+platforms such as GitHub.
+
+If you haven’t already done so, you’ll need to create a (free) GitHub account
+on http://github.com and have Git tools available on your development system.
+
+Repository layout
+*****************
+
+To clone the main Zephyr Project repository use::
+
+    $ git clone https://github.com/zephyrproject-rtos/zephyr
+
+The Zephyr project directory structure is described in `Source Tree Structure`_
+documentation. In addition to the Zephyr kernel itself, you'll also find the
+sources for technical documentation, sample code, supported board
+configurations, and a collection of subsystem tests.  All of these are
+available for developers to contribute to and enhance.
+
+.. _Source Tree Structure:
+   https://www.zephyrproject.org/doc/kernel/overview/source_tree.html
+
+Pull Requests and Issues
+************************
+
+.. _Zephyr Project Issues: https://jira.zephyrproject.org
+
+.. _open pull requests: https://github.com/zephyrproject-rtos/zephyr/pulls
+
+Before starting on a patch, first check in our Jira `Zephyr Project Issues`_
+system to see what’s been reported on the issue you’d like to address.  Have a
+conversation on the mailing list (or IRC) to see what others think of your
```

Comment:
I still think it'd be better to have links to the mailing list and IRC here, but I'm OK merging this without them.